prefect_gcp/workers/vertex.py

```diff
@@ -0,0 +1,633 @@
+            "network": "{{ network }}",
+            "reserved_ip_ranges": "{{ reserved_ip_ranges }}",
+            "maximum_run_time_hours": "{{ maximum_run_time_hours }}",
+            "worker_pool_specs": [
+                {
+                    "replica_count": 1,
+                    "container_spec": {
+                        "image_uri": "{{ image }}",
+                        "command": "{{ command }}",
+                        "args": [],
+                    },
+                    "machine_spec": {
+                        "machine_type": "{{ machine_type }}",
+                        "accelerator_type": "{{ accelerator_type }}",
+                        "accelerator_count": "{{ accelerator_count }}",
+                    },
+                    "disk_spec": {
+                        "boot_disk_type": "{{ boot_disk_type }}",
+                        "boot_disk_size_gb": "{{ boot_disk_size_gb }}",
+                    },
+                }
+            ],
+        }
+    )
+    job_watch_poll_interval: float = Field(
+        default=5.0,
+        title="Poll Interval (Seconds)",
+        description=(
+            "The amount of time to wait between GCP API calls while monitoring the "
+            "state of a Vertex AI Job."
+        ),
+    )
+
+    @property
+    def project(self) -> str:
+        """property for accessing the project from the credentials."""
+        return self.credentials.project
+
+    @property
+    def job_name(self) -> str:
+        """
+        The name can be up to 128 characters long and can be consist of any UTF-8 characters. Reference:
+        https://cloud.google.com/python/docs/reference/aiplatform/latest/google.cloud.aiplatform.CustomJob#google_cloud_aiplatform_CustomJob_display_name
+        """  # noqa
+        job_name_base = self.job_spec["name"]
+        unique_suffix = uuid4().hex
+        job_name = f"{job_name_base}-{unique_suffix}"
+        return job_name
+
+    def prepare_for_flow_run(
+        self,
+        flow_run: "FlowRun",
+        deployment: Optional["DeploymentResponse"] = None,
+        flow: Optional["Flow"] = None,
+    ):
+        super().prepare_for_flow_run(flow_run, deployment, flow)
+
+        self._ensure_at_least_one_worker_pool_spec()
+        self._inject_name_if_not_present()
+        self._inject_formatted_env_vars()
+        self._inject_formatted_command()
+        self._ensure_existence_of_service_account()
+        self._ensure_existence_of_container_image()
+
+    def _ensure_at_least_one_worker_pool_spec(self):
+        """Ensures that at least one worker pool spec is defined"""
+
+        provided_worker_pool_specs = self.job_spec.get("worker_pool_specs", [])
+        if len(provided_worker_pool_specs) == 0:
+            raise ValueError(
+                "At least one worker pool spec is required for the Vertex job. "
+                "Please pass in a valid worker pool spec."
+            )
+
+    def _inject_name_if_not_present(self):
```

Comment:
I don't think this method is necessary since `name` isn't an accepted field on `CustomJobSpec` according to the [docs](https://cloud.google.com/python/docs/reference/aiplatform/latest/google.cloud.aiplatform_v1.types.CustomJobSpec).

Reply:
eh, i called it `name` here, but it's actually being used for `CustomJob.display_name`, which appears to be a required field

https://cloud.google.com/python/docs/reference/aiplatform/latest/google.cloud.aiplatform_v1.types.CustomJob

![image](https://github.com/PrefectHQ/prefect-gcp/assets/19556538/f1c86953-721c-465c-8202-80750d49c88d)

Reply:
maybe `name` shouldn't be placed into the job spec 🤔 let me actually remove this from this template expression and just key off of it directly from the base class

Reply:
ok yeah idk what i was thinking here, so i removed the `{{ name }}` from the job_spec template and just going to use the inherited `.name` property